kieker-monitoring/test/kieker/test/monitoring/junit/writer/collector/TestChunkingCollector.java

```diff
@@ -0,0 +1,121 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+package kieker.test.monitoring.junit.writer.collector;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.locks.LockSupport;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import kieker.common.configuration.Configuration;
+import kieker.common.record.controlflow.OperationExecutionRecord;
+import kieker.monitoring.core.configuration.ConfigurationFactory;
+import kieker.monitoring.core.controller.IMonitoringController;
+import kieker.monitoring.core.controller.MonitoringController;
+import kieker.monitoring.writer.collector.ChunkingCollector;
+import kieker.monitoring.writer.serializer.StringSerializer;
+
+/**
+ * Basic tests for the chunking record collector.
+ *
+ * @author holger
+ *
+ * @since 1.13
+ */
+public final class TestChunkingCollector {
+
+	private static final Charset CHARSET = Charset.forName("UTF-8");
+	
+	public TestChunkingCollector() {
+		// Default Constructor
+	}
+	
+	/**
+	 * Tests both writing of chunks due to size (a full chunk can be written) and to time (records remain unwritten for some time).
+	 *
+	 * @throws IOException
+	 *             Not expected
+	 */
+	@Test
+	public void testChunkingSizeAndTime() throws IOException {
+		final String testId = "testChunkingSizeAndTime";
+		final int recordCount = 25;
+		final int deferredWriteDelay = 500;
+
+		final IMonitoringController controller = this.createController(testId, deferredWriteDelay);
+
+		for (int recordIndex = 0; recordIndex < recordCount; recordIndex++) {
+			final OperationExecutionRecord record = new OperationExecutionRecord("op()", "SESS-ID", 0, recordIndex, recordIndex, "host", recordIndex, 1);
+			controller.newMonitoringRecord(record);
+		}
+
+		// Wait until a timed write should have occurred
+		LockSupport.parkNanos(2 * deferredWriteDelay * 1000000L);
+		controller.terminateMonitoring();
+
+		// Retrieve written data from the data storage
+		final byte[] data = TestRawDataStorage.getInstance().getData(testId);
+		final List<String> lines = TestChunkingCollector.linesFromData(data);
+
+		// Count the number of lines (= records) written
+		final int numberOfLines = lines.size();
+
+		// One more line due to the metadata record
+		Assert.assertEquals(recordCount + 1, numberOfLines);
+	}
+
+	private static List<String> linesFromData(final byte[] data) throws IOException {
+		final List<String> lines = new ArrayList<String>();
+
+		final BufferedReader reader = new BufferedReader(new InputStreamReader(new ByteArrayInputStream(data), CHARSET));
```

Comment:
Since Kieker now requires Java 7, you can use the try-with-resource clause here.